network/src/lib.rs

```diff
@@ -835,15 +834,15 @@ impl PolkadotProtocol {
 		targets: HashSet<ValidatorId>,
 		collation: Collation,
 		outgoing_targeted: OutgoingMessages,
-	) -> impl futures::future::Future<Item = (), Error=()> {
+	) -> impl futures::future::Future<Output = ()> {
```

Comment:
Not sure if this PR is the right place to do it, but probably this function could just become `async` now, so the `Either` could also go away.